graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrNodeImpl.java

```diff
@@ -562,6 +579,25 @@ public String getThumbnailUrl(@GraphQLName("name") @GraphQLDescription("Thumbnai
         return node.getThumbnailUrl(name);
     }
 
+    @Override
+    @GraphQLDescription("Get render URL")
+    public String getRenderUrl(@GraphQLName("workspace") @GraphQLDescription("The target workspace") @GraphQLNonNull NodeQueryExtensions.Workspace workspace,
+                               @GraphQLName("language") @GraphQLDescription("The language content is rendered in") @GraphQLNonNull String language,
+                               @GraphQLDefaultValue(GqlUtils.SupplierFalse.class) @GraphQLName("findDisplayable") @GraphQLDescription("Finds displayable node") Boolean findDisplayable,
+                               DataFetchingEnvironment environment) {
+        try {
+            String url = getNodeURL(this.node, workspace.getValue(), LanguageCodeConverters.languageCodeToLocale(language), findDisplayable);
+            HttpServletResponse httpServletResponse = ContextUtil.getHttpServletResponse(environment.getGraphQlContext());
+            HttpServletRequest httpServletRequest = ContextUtil.getHttpServletRequest(environment.getGraphQlContext());
+            url = urlRewriteService.rewriteOutbound(url, httpServletRequest, httpServletResponse);
+            // Need to strip off /modules as it is being added after rewrite rules are applied. It is not needed for the
+            // final url.
+            return url.replace("/modules", "");
```

Comment:
The current replace code still look dangerous to me, as it is arbitrary stripping the String `/modules` chain from the url, no matter the place of this chain.

Are you 100% that the /modules that will be replace here will always be the one you want to remove ? Do we risk anything for node paths like this one:
`/sites/digitall/home/pageA/area/modulesPresentation/moduleA`

Reply:
One option, that could be interesting to try would consist on wrapping the currentRequest and override the .getPath to return `/cms` instead of `/modules`. 
The goal would be to make the current request done on `/modules/**` like if it was done on `/cms/**` transparently by manipulating/wrapping the request/response objects.
Not sure if it's easy or not, but just wanted to share the idea.

Reply:
Wrapper looks good indeed.

Reply:
Update, indeed the same issue happened again in the past:
Take a look at this GraphQL field: `RenderNodeExtensions.getRenderedContent(...)`
and the chosen solution seem's to be: https://github.com/Jahia/graphql-core/blob/master/graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/render/RenderNodeExtensions.java#L144-L146

simply unwrap the original request from the OSGI proxy, to get access to the original request before it was transformed for the OSGI GraphQL servlet by our ServletHandler.

Could you please update your implem to replace the wrapper by the same solution from: `RenderNodeExtensions.getRenderedContent(...)` ?
Also it could be interesting to create an utility function somewhere to avoid code duplicate, considering the troubles we had to found back this solution, it could be interesting to put it somewhere useful with some code comments for the future.
And no need for additional tech ticket, current behavior is correct, and at any time we can have access to the original request.